File: SS/Core/Service/src/ca/sciencestudio/service/utilities/ModelPathUtils.java
/** Copyright (c) Canadian Light Source, Inc. All rights reserved.
 *   - see license.txt for details.
 *
 *  Description:
 *     ModelPathUtils class.
 *     
 */
package ca.sciencestudio.service.utilities;

/**
 * @author maxweld
 *
 */
public abstract class ModelPathUtils {
	
	public static final String PERSON_PATH = "/persons";
	public static final String PROJECT_PATH = "/projects";
	public static final String PROJECT_PERSON_PATH = "/project/persons";
	public static final String SESSION_PATH = "/sessions";
	public static final String SESSION_PERSON_PATH = "/session/persons";
	public static final String EXPERIMENT_PATH = "/experiments";
	public static final String SAMPLE_PATH = "/samples";
	public static final String SCAN_PATH = "/scans";
	
	public static String TREE_PATH = "/ss/tree";
	public static String MODEL_PATH = "/ss/model";
	
	public static void setTreePath(String treePath) {
		TREE_PATH = treePath;
	}
	public static void setModelPath(String modelPath) {
		MODEL_PATH = modelPath;
	}
	
	public static String getTreeProjectPath(String... exts) {
		return join(TREE_PATH, PROJECT_PATH, exts);
	}
	public static String getModelProjectPath(String... exts) {
		return join(MODEL_PATH, PROJECT_PATH, exts);
	}

	public static String getTreeProjectPersonPath(String... exts) {
		return join(TREE_PATH, PROJECT_PERSON_PATH, exts);
	}
	public static String getModelProjectPersonPath(String... exts) {
		return join(MODEL_PATH, PROJECT_PERSON_PATH, exts);
	}
	
	public static String getTreeSessionPath(String... exts) {
		return join(TREE_PATH, SESSION_PATH, exts);
	}
	public static String getModelSessionPath(String... exts) {
		return join(MODEL_PATH, SESSION_PATH, exts);
	}
	
	public static String getTreeSessionPersonPath(String... exts) {
		return join(TREE_PATH, SESSION_PERSON_PATH, exts);
	}
	public static String getModelSessionPersonPath(String... exts) {
		return join(MODEL_PATH, SESSION_PERSON_PATH, exts);
	}
	
	public static String getTreeExperimentPath(String... exts) {
		return join(TREE_PATH, EXPERIMENT_PATH, exts);
	}
	public static String getModelExperimentPath(String... exts) {
		return join(MODEL_PATH, EXPERIMENT_PATH, exts);
	}
	
	public static String getTreeSamplePath(String... exts) {
		return join(TREE_PATH, SAMPLE_PATH, exts);
	}
	public static String getModelSamplePath(String... exts) {
		return join(MODEL_PATH, SAMPLE_PATH, exts);
	}
	
	public static String getTreeScanPath(String... exts) {
		return join(TREE_PATH, SCAN_PATH, exts);
	}
	public static String getModelScanPath(String... exts) {
		return join(MODEL_PATH, SCAN_PATH, exts);
	}
	
	protected static String join(String prefix, String path, String... exts) {
		StringBuffer buffer = new StringBuffer();
		buffer.append(prefix).append(path);
		for(String e : exts) {
			buffer.append(e);
		}
		return buffer.toString();
	}
}

File: SS/Core/WebApp/WebContent/js/main/modelPathUtils.js
/**	Copyright (c) Canadian Light Source, Inc. All rights reserved.
 *   - see license.txt for details.
 *
 *  Description:
 *     Model path utilities.
 *     
 */

var ModelPathUtils = {

	PERSON_PATH:"/persons",
	
	PROJECT_PATH:'/projects',

    PROJECT_PERSON_PATH:'/project/persons',
	
    SESSION_PATH:'/sessions',
    
    SESSION_PERSON_PATH:'/session/persons',
    
    EXPERIMENT_PATH:"/experiments",
    
    SAMPLE_PATH:'/samples',
    
    SCAN_PATH:"/scans",
    
    MODEL_PATH:'/ss/model',
    
    getPersonPath: function(exts) {
    	return ModelPathUtils.specialJoin(ModelPathUtils.MODEL_PATH + ModelPathUtils.PERSON_PATH, exts);	
	},
    
	getModelProjectPath: function(exts) {
		return ModelPathUtils.specialJoin(ModelPathUtils.MODEL_PATH + ModelPathUtils.PROJECT_PATH, exts);
	},
	
	getModelProjectPersonPath: function(exts) {
		return ModelPathUtils.specialJoin(ModelPathUtils.MODEL_PATH + ModelPathUtils.PROJECT_PERSON_PATH, exts);
	},
	
	getModelSessionPath: function(exts) {
		return ModelPathUtils.specialJoin(ModelPathUtils.MODEL_PATH + ModelPathUtils.SESSION_PATH, exts);
	},
	
	getModelSessionPersonPath: function(exts) {
		return ModelPathUtils.specialJoin(ModelPathUtils.MODEL_PATH + ModelPathUtils.SESSION_PERSON_PATH, exts);
	},
	
	getModelExperimentPath: function(exts) {
		return ModelPathUtils.specialJoin(ModelPathUtils.MODEL_PATH + ModelPathUtils.EXPERIMENT_PATH, exts);
	},
	
	getModelSamplePath: function(exts) {
		return ModelPathUtils.specialJoin(ModelPathUtils.MODEL_PATH + ModelPathUtils.SAMPLE_PATH, exts);
	},
	
	getModelScanPath: function(exts) {
		return ModelPathUtils.specialJoin(ModelPathUtils.MODEL_PATH + ModelPathUtils.SCAN_PATH, exts);
	},
	
	specialJoin: function(prefix, exts) {
		var path = prefix;
		if(exts) {
			if(exts.join) { 
				path += exts.join('');
			} else { 
				path += exts;
			}
		}
		return path;
	}
};


